pkg/cloud/gcp/gcs_kms.go

```diff
@@ -81,26 +93,37 @@ func MakeGCSKMS(uri string, env cloud.KMSEnv) (cloud.KMS, error) {
 			)
 		}
 
-		// Credentials are passed in base64 encoded, so decode the credentials first.
-		credentialsJSON, err := base64.StdEncoding.DecodeString(kmsURIParams.credentials)
-		if err != nil {
-			return nil, err
-		}
-
-		credentialsOpt = append(credentialsOpt, option.WithCredentialsJSON(credentialsJSON))
+		clientOpts = append(clientOpts, credentialsOpt)
 	case cloud.AuthParamImplicit:
 		if env.KMSConfig().DisableImplicitCredentials {
 			return nil, errors.New(
 				"implicit credentials disallowed for gcs due to --external-io-implicit-credentials flag")
 		}
 		// If implicit credentials used, no client options needed.
+	case cloud.AuthParamAssume:
+		if kmsURIParams.target == "" {
+			return nil, errors.Errorf(
+				"%s is set to '%s', but %s is not set",
+				cloud.AuthParam,
+				cloud.AuthParamAssume,
+				TargetPrincipalParam,
+			)
+		}
+		impersonateOpt, err := createImpersonateCredentials(ctx, kmsURIParams.target, kms.DefaultAuthScopes(), kmsURIParams.credentials)
+		if err != nil {
+			return nil, errors.Wrapf(err, "error impersonating %s", kmsURIParams.target)
+		}
+
+		clientOpts = append(
+			clientOpts,
+			impersonateOpt,
+			option.WithScopes(kms.DefaultAuthScopes()...),
```

Comment:
just noting that in gcs_storage.go when we are setting up the client we don't seem to add an `option.WithScopes` like we do here.